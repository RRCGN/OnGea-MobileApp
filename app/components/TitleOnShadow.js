/**
 * Title on a Inset Shadow.
 * Has a big Title and maybe a smaller Subtitle.
 * @flow
 */

import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'


type Props = {
  title: string,
  subtitle?: ?string
}

const TitleOnShadow = ({ title, subtitle }: Props) => (
  <LinearGradient
    colors={['rgba(0,0,0,0)', 'rgba(0,0,0,0.8)']}
    style={styles.overlay}
  >
    <Text style={[styles.text, styles.title]}>{title}</Text>
    {subtitle && <Text style={[styles.text, styles.subtitle]}>{subtitle}</Text>}
  </LinearGradient>
)

export default TitleOnShadow


const styles = StyleSheet.create({
  overlay: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    paddingTop: 24,
    paddingBottom: 16,
    paddingLeft: 16,
    paddingRight: 16
  },
  text: {
    color: 'white',
    backgroundColor: 'transparent'
  },
  title: {
    fontSize: 24
  },
  subtitle: {
    fontSize: 14
  },
})
