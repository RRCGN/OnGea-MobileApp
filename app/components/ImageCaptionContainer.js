/**
 * Container for an Image with a Caption
 * @flow
 */

import React from 'react'
import {
  View,
  StyleSheet
} from 'react-native'
import TitleOnShadow from './TitleOnShadow'


type Props = {
  image: ReactElement<Image>,
  caption: ReactElement<TitleOnShadow>
}

const ImageCaptionContainer = ({ image, caption }: Props) => (
  <View style={{ flex: 1, position: 'relative' }}>
    {image}
    {caption}
  </View>
)

export default ImageCaptionContainer
