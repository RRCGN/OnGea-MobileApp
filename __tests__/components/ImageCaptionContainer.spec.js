import { View } from 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import TitleOnShadow from '../../app/components/TitleOnShadow'
import ImageCaptionContainer from '../../app/components/ImageCaptionContainer'

import renderer from 'react-test-renderer'

describe('<ImageCaptionContainer />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <ImageCaptionContainer
        caption={<TitleOnShadow title="Foobar" />}
        source={{ uri: 'http://placehold.it/1x1'}}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
