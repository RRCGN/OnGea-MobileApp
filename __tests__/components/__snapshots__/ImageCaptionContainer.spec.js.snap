exports[`<ImageCaptionContainer /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }>
  <Image
    source={
      Object {
        "uri": "http://placehold.it/1x1",
      }
    }
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "height": undefined,
        "resizeMode": "cover",
        "width": undefined,
      }
    } />
  <TitleOnShadow
    title="Foobar" />
</View>
`;
